x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -288,39 +297,86 @@ protected LogicalPlan rule(UnaryPlan plan) {
+                    var tuple = projectAggregations(project.projections(), aggs);
+                    // project can be fully removed
+                    if (tuple.v1().isEmpty()) {
+                        var newAggs = tuple.v2();
+                        var newGroups = replacePrunedAliasesUsedInGroupBy(a.groupings(), aggs, newAggs);
+                        plan = new Aggregate(a.source(), a.child(), newGroups, newAggs);
+                    }
                 }
+                return plan;
             }
 
             // Agg with underlying Project (group by on sub-queries)
             if (plan instanceof Aggregate a) {
                 if (child instanceof Project p) {
-                    return new Aggregate(a.source(), p.child(), a.groupings(), combineProjections(a.aggregates(), p.projections()));
+                    plan = new Aggregate(a.source(), p.child(), a.groupings(), combineProjections(a.aggregates(), p.projections()));
                 }
             }
 
             return plan;
         }
 
+        // variant of #combineProjections specialized for project followed by agg due to the rewrite rules applied on aggregations
+        // this method tries to combine the projections by paying attention to:
+        // - aggregations that are projected away - remove them
+        // - aliases in the project that point to aggregates - keep them in place (to avoid duplicating the aggs)
+        private Tuple<List<? extends NamedExpression>, List<? extends NamedExpression>> projectAggregations(
+            List<? extends NamedExpression> upperProjection,
+            List<? extends NamedExpression> lowerAggregations
+        ) {
+            AttributeMap<Expression> lowerAliases = new AttributeMap<>();
+            for (NamedExpression ne : lowerAggregations) {
+                lowerAliases.put(ne.toAttribute(), Alias.unwrap(ne));
+            }
+
+            // flag indicating if the projection introduces at least one extra alias
+            boolean keepProject = false;
+            AttributeSet seen = new AttributeSet();
+            for (NamedExpression upper : upperProjection) {
+                Expression unwrapped = Alias.unwrap(upper);
+                // projection contains an inner alias (point to an existing fields inside the projection)
+                if (seen.contains(unwrapped)) {
+                    keepProject = true;
+                    break;
+                }
+                seen.add(Expressions.attribute(unwrapped));
+            }
+
+            // remove lower aggs that are not referenced in the upper projections
+            List<? extends NamedExpression> keptProjections = upperProjection;
+            if (keepProject == false) {
+                keptProjections = emptyList();
+                lowerAggregations = combineProjections(upperProjection, lowerAggregations);
+            }
+
+            return new Tuple<>(keptProjections, lowerAggregations);
```

Comment:
the Tuple can return a bool directly, since the `keptProjections` are never used anyways.

Reply:
It is:
```
                   if (tuple.v1().isEmpty()) {
                        var newAggs = tuple.v2();
                        var newGroups = replacePrunedAliasesUsedInGroupBy(a.groupings(), aggs, newAggs);
                        plan = new Aggregate(a.source(), a.child(), newGroups, newAggs);
                    }
```